cpp/src/parquet/column_writer.cc

```diff
@@ -2299,12 +2319,112 @@ Status TypedColumnWriterImpl<FLBAType>::WriteArrowDense(
   return Status::OK();
 }
 
+template <typename DType>
+void TypedColumnWriterImpl<DType>::UpdateBloomFilter(const T* values,
+                                                     int64_t num_values) {
+  if (bloom_filter_) {
+    for (int64_t i = 0; i < num_values; ++i) {
+      bloom_filter_->InsertHash(bloom_filter_->Hash(values + i));
```

Comment:
This will be very slow in any case, because there is a double virtual indirection (first `bloom_filter_->Hash`, then `hasher_->Hash`).

I think you want to add batch methods, such as `BloomFilter::InsertHashes(const double* values, int64_t num_values)`.

Reply:
Emmmm, I'll try to seperate a patch for batch methods